go/vt/tabletmanager/state_change.go

```diff
@@ -293,6 +293,27 @@ func (agent *ActionAgent) changeCallback(ctx context.Context, oldTablet, newTabl
 		}
 	}
 
+	// See if we need to enable or disable semi-sync replication.
+	// Note we never enable the master side, as that would have
+	// been done by either the reparent process, or an external
+	// tool (like Orchestrator).
+	if *enableSemiSync {
```

Comment:
So what's preventing us from making ActionAgent.enableSemiSync to be for all transitions? And actually here in this case we don't really have to verify whether we actually transitioned or not, we can just set/reset semi-sync to be compliant with what we want, right? If it won't change anything and will just stop/start slave, that won't hurt.